### PR TITLE
Account migration

### PR DESCRIPTION
**Pull request by arthru**
End of @alexis-via's work. See #24

### CODE REVIEW

**Comment by bwrsandman** [comment]
This PR has conflicts. Could you rebase it?

**Comment by arthru** [comment]
Rebased !

File: addons/account/migrations/8.0.1.1/openupgrade_analysis_work.txt [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+account      / analytic.entries.report  / month (selection)             : DEL selection_keys: ['01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11', '12']
+account      / analytic.entries.report  / year (char)                   : DEL 
+# Nothing to do : analytic.entries.report is a view
+
+account      / res.partner              / invoice_ids (one2many)        : relation is now 'account.invoice' ('account.invoice.line')
+# Nothing to do for this field : partner_id on invoice_line is related from invoi
+
+---XML records in module 'account'---
+NEW account.analytic.journal: account.exp
+## This was a demo data that became a data ; update in pre-migration
```
Comment (bwrsandman) [inline addons/account/migrations/8.0.1.1/openupgrade_analysis_work.txt:56]: Does this work in dbs with and without demo data?

File: addons/account/migrations/8.0.1.1/openupgrade_analysis_work.txt [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+# Kept as a legacy column
+
+account      / account.bank.statement.line / amount_currency (float)       : NEW 
+# Nothing to do : optional field in the case the bank statement is in another currency
+
+account      / account.bank.statement.line / analytic_account_id (many2one): DEL relation: account.analytic.account
+# Kept as a legacy column
+
+account      / account.bank.statement.line / bank_account_id (many2one)    : NEW relation: res.partner.bank
+# TODO is there a way to initialize that field ?
```
Comment (bwrsandman) [inline addons/account/migrations/8.0.1.1/openupgrade_analysis_work.txt:13]: Could this just be a rename of account_id from L6?

File: addons/account/migrations/8.0.1.1/openupgrade_analysis_work.txt (hunk 1 above)
Comment (arthru) [inline addons/account/migrations/8.0.1.1/openupgrade_analysis_work.txt:56]: Yes, check lines 45 to 50 in pre-migration.py

Comment (legalsylvain) [inline addons/account/migrations/8.0.1.1/openupgrade_analysis_work.txt:56]: Maybe i'm not very up to date, but I thought that "OpenUpgrade + Demo data" didn't work for other reasons.

Comment (bwrsandman) [inline addons/account/migrations/8.0.1.1/openupgrade_analysis_work.txt:56]: It was the case fro 7.0, but during the code sprint I was using demo data with no problem.
The issue, I think is just the link in ir.model.data, which means having it in production is a bad idea.
For developement, though, very useful.

File: addons/account/migrations/8.0.1.1/openupgrade_analysis_work.txt (hunk 2 above)
Comment (arthru) [inline addons/account/migrations/8.0.1.1/openupgrade_analysis_work.txt:13]: No, this new field is a many2one to a res.partner.bank whereas line 6 removal was a many2one to an account.account

Comment (bwrsandman) [inline addons/account/migrations/8.0.1.1/openupgrade_analysis_work.txt:13]: You may be able to get the bank from the account.journal associated with the old account.

Comment (StefanRijnhart) [inline addons/account/migrations/8.0.1.1/openupgrade_analysis_work.txt:13]: No, it's a different thing. Apart from that, bank_account_id is used nowhere at all.

**Comment by bwrsandman** [comment]
:+1:
Points of worry explained. Looks good to me.

**Comment by StefanRijnhart** [comment]
I'm repeating the same concern as in another PR: now contains a commit from the leading 8.0 branch (adding project scripts) which was reverted later on. Anyone knows what the way is to deal with this?

**Comment by arthru** [comment]
I'm going to clean up that pull request.

**Comment by arthru** [comment]
Ok, the others commit have been removed.
Enjoy the merge ;)

**Comment by legalsylvain** [comment]
@arthru : what did you done ? 
I have the same problem in that PR #78 

Regards.

**Comment by diellenne** [comment]
Hi,

who is working on this merge and when is planned to be done?
If anyone if taking care of it I'd really like to help but to do this I should be added to the project collaborators...

Greetings,
D

**Comment by dreispt** [comment]
@diellenne actually on GitHub anyone can collaborate with a project: you just need to fork, code and then make a Pull Request to the main repository.

**Comment by diellenne** [comment]
@dreispt thank you, I will act in that way to merge my modifications and the migration scripts we are implementing in our fork but this is already a 'Pull request'; so I think is a bit useless to make the 'accounting' merge on my own branch and then make another 'Pull request' to this repository...

**Comment by dreispt** [comment]
@diellenne Both approaches are viable; I guess it depends on the expectation for this to be merged. I would merge this PR in my work branch and make my changes there.

**Comment by plaes** [comment]
What's currently blocking this PR?

File: addons/account/migrations/8.0.1.1/openupgrade_analysis_work.txt [hunk 3]
```diff
@@ -0,0 +1,189 @@
+---Fields in module 'account'---
+
+account      / account.bank.statement   / message_last_post (datetime)  : NEW 
+# Field initialized
+
+account      / account.bank.statement.line / account_id (many2one)         : DEL relation: account.account, required: required
```
Comment (alhashash) [inline addons/account/migrations/8.0.1.1/openupgrade_analysis_work.txt]: `account_id` was added again in commit 5ae60d8a2d432cf3280487743b9f413f6ef27e08 to fix PoS issue #932 and is actually used to generate the journal entry. We still need to test if draft bank statement in v7 can be correctly validated after migration to v8.

Comment (legalsylvain) [inline addons/account/migrations/8.0.1.1/openupgrade_analysis_work.txt]: Hi @alhashash, 
As I see in V7, account_id is present in account.bank.statement.line model. (and Finally present in V8 as you said)
Ref: https://github.com/odoo/odoo/blob/7.0/addons/account/account_bank_statement.py#L560
So I guess that there is "nothing to do".
Can you detail your point of you ?
Regards.

File: addons/account/migrations/8.0.1.1/pre-migration.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.openupgrade import openupgrade
+
+column_renames = {
+    'account_bank_statement_line': [
+        ('account_id', None),
```
Comment (legalsylvain) [inline addons/account/migrations/8.0.1.1/pre-migration.py]: @arthru: as @alhashash said, it is now unnecessary as account_id is present in V8.

**Comment by arthru** [comment]
I just updated the code with the latest comments.

File: addons/account/migrations/8.0.1.1/openupgrade_analysis_work.txt (hunk 3 above)
Comment (alhashash) [inline addons/account/migrations/8.0.1.1/openupgrade_analysis_work.txt]: @legalsylvain
The `account_id` field was removed initially from v8, so, the analysis shows it as deleted. Then, it was reintroduced again in the mention commit.  arthru fixed in commit e9db83f11e420ceb127a8f985de2e9d067ae9aab.

**Comment by legalsylvain** [comment]
Thanks a lot.
:+1:

**Pull request merged**